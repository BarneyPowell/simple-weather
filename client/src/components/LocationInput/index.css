.location-input {
    display: flex;
    flex-direction: column;
    padding: 1em;
}

.location-input label {
    flex: 0 1 auto;
    margin-bottom: 0.5em;
    font-size: 0.8em;
}

.location-input input {
    flex: 0 1 auto;
    border: none;
    border-bottom: 2px solid #abcdef;
    background-color: #f9f9f9;
    padding: 0.5em;
    font-size: 1em;
}