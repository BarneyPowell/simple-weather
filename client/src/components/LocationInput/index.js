import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './index.css';

const KeyCodes = {
    ENTER: 13
}

class LocationInput extends Component
{

    constructor(props) {
        super(props);

        this.handleKeyPress = this.handleKeyPress.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleKeyPress(e) {
        if(e.which === KeyCodes.ENTER
            && this.props.onAction
            && typeof this.props.onAction === 'function') 
            {
                this.props.onAction(e.target.value);
            }
    }

    handleChange(e) {
        if(this.props.onChange && typeof this.props.onChange === 'function') {
            this.props.onChange(e.target.value);
        }
    }

    render() {
        return (
            <div className="location-input">
                <label>{this.props.label}</label>
                <input type="text" 
                    onKeyPress={this.handleKeyPress} 
                    onChange={this.handleChange}
                    placeholder="City name" />
            </div>
        );
    }
}

LocationInput.propTypes = {
    label: PropTypes.string,
    onChange: PropTypes.func,
    onAction: PropTypes.func
}

export default LocationInput;