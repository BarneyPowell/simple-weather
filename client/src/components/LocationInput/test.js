import React from 'react';
import LocationInput from './';

it('should render without crashing', () => {
    const wrapper = shallow(<LocationInput />);
    expect(wrapper).toMatchSnapshot();
});

it('should set the label to the appropriate value', () => {
    const wrapper = mount(<LocationInput label="test" />);
    expect(wrapper.find("label").text()).toEqual("test");
});