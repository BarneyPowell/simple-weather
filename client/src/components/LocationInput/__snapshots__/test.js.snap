// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="location-input"
>
  <label />
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="City name"
    type="text"
  />
</div>
`;
